File: spring-best/src/main/java/com/best/springbest/model/ProductSize.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "product_size")
public class ProductSize {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id",nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "size_id",nullable = false)
    private Size size;
}

File: BEST-App-main/lib/model/product_size.dart
class ProductSize {
  final int id;
  final int productId;
  final int sizeId;

  ProductSize(this.id, this.productId, this.sizeId);

  static Map<String, dynamic> toJson(ProductSize productSize) {
    return {
      'id': productSize.id,
      'product_id': productSize.productId,
      'size_id': productSize.sizeId,
    };
  }
}
